Release 1.0

ricsantos opened the pull request:
Brings ManUp inline with [ionic-manup](https://github.com/NextFaze/ionic-manup) by:

- updating the config.json format, now has (optional) nested `ios` object, and newer, friendlier keys `latest`, `minimum`, `url`, etc... #15 #11 
- add support for the `enabled` key, aka maintenance mode #16

ricsantos committed:
Merge branch 'release/0.7' into develop

ricsantos committed:
Use the same json format as ionic-manup #15

ricsantos committed:
Wrap keys in (optional) `ios` json object #15

ricsantos committed:
Add maintenance mode (app is enabled) check #16

ricsantos committed:
Keep track of the maintenance alert and hide it if the app's enabled

swoolcock commented on ManUp/ManUp.m:
```diff
@@ -78,6 +79,7 @@ - (id)init {
 	if (self = [super init]) {
 		self.updateInProgress = NO;
```
Tabs D: D: D:

ricsantos replied:
😱

swoolcock commented on ManUp/ManUp.m:
```diff
@@ -265,10 +282,31 @@ - (void)checkAppVersion {
+            self.alertController = nil;
+            
+        } else if (!enabled && !self.maintenanceModeShown) {
+            [self showAlertWithTitle:[NSString stringWithFormat:NSLocalizedString(@"%@ Unavailable", nil), appName]
```
Comments in `NSLocalizedString` please. 💩

ricsantos replied:
Will fix 👍

awulf commented on ManUp/ManUp.m:
```diff
@@ -265,10 +282,31 @@ - (void)checkAppVersion {
     NSComparisonResult currentVersionComparisonResult = [ManUp compareVersion:installedVersion toVersion:currentVersion];
     
-    if (self.alertController) {
+    if (self.alertController && !self.maintenanceModeShown) {
```
Maintenance mode should be shown in favour of an update alert. That is an update alert should be removed snd replaced with a maintenance alert

ricsantos replied:
Good point. I will change it so that it always removes a current alert before showing a new one.

awulf commented on ManUp/ManUp.m:
```diff
@@ -265,10 +282,31 @@ - (void)checkAppVersion {
+            
+        } else if (!enabled && !self.maintenanceModeShown) {
+            [self showAlertWithTitle:[NSString stringWithFormat:NSLocalizedString(@"%@ Unavailable", nil), appName]
+                             message:[NSString stringWithFormat:NSLocalizedString(@"%@ is currently unavailable. Please check back later.", nil), appName]
```
Add comments to localized strings where parameters are given eg "{appName} Unavailable"

ricsantos replied:
Will fix 💃

ricsantos committed:
Always show the maintenance mode alert even if another alert is shown

ricsantos reviewed (commented):
I didn't mean to start a review here.

ricsantos committed:
Add localized string descriptions

ricsantos committed:
Remove the shared instance property

The pattern of a singleton with a delegate is a bit gross. This removes
the singleton property, and adds a constructor that takes the url and a
delegate.

Bonus: this refactor removes the default dictionary/json file as it is
rarely (never) used in practice.

ricsantos committed:
Rename serverConfigURL to configURL

ricsantos committed:
Update the readme

ricsantos committed:
Update the readme

ricsantos committed:
Merge pull request #18 from NextFaze/remove-singleton

Remove singleton

ricsantos committed:
Update the podspec
